### PR TITLE
updated RELEASE_NOTES with recent PRs that have been merged

### PR DESCRIPTION
**Pull request by edwardhartnett**
Release notes updates.

### CODE REVIEW

**Comment by ckhroulev** [comment]
I think it would make sense to state that #350 is fixed in 4.7.3 and not in 4.6.2-rc2. This would make it easier to see what range of versions is affected by this issue.

**Comment by edwardhartnett** [comment]
@ckhroulev I agree. Stand by and I will add it to this PR.

**Comment by WardF** [comment]
Merging into release wellspring branch, will find its way back upstream to `master` shortly.

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ This file contains a high-level description of this package's evolution. Release
 
 ## 4.7.3 - TBD
 
+* [Bug Fix] Library was modified so that rewriting the same attribute happens without deleting the attribute, to avoid a limit on how many times this may be done in HDF5. This fix was thought to be in 3.6.2 but was not. See [https://github.com/Unidata/netcdf-c/issues/350].
```
Comment (ckhroulev) [inline RELEASE_NOTES.md]: @WardF Please note that this entry is not correct. It should say 4.6.2, not 3.6.2.

**Pull request merged**